File: src/lib/committeeDuty.ts
export const STANDARD_DUTY_PRESETS = [
  'Maintenance & Repairs',
  'Security & Gate',
  'Housekeeping',
  'Electrical',
  'Plumbing',
  'Garden & Landscaping',
  'Lift / Elevator',
  'Parking',
  'Accounts & Billing',
  'Event Coordination',
] as const;

export type CommitteeDutiesChartRow = {
  id: string;
  society_id: string;
  period_from: string;
  period_to: string | null;
  is_active: boolean;
};

export type CommitteeDutyRow = {
  id: string;
  chart_id: string;
  duty_label: string;
  supervisor_names: string[];
  sort_order: number;
};

export type DutyFormRow = {
  id: string | null;
  dutyLabel: string;
  supervisorNames: string;
  sortOrder: number;
};

export function parseSupervisorNames(input: string): string[] {
  return input
    .split(/[,;]+/)
    .map((s) => s.trim())
    .filter(Boolean);
}

export function formatSupervisorNames(names: string[]): string {
  return names.filter(Boolean).join(', ');
}

export function dutiesPeriodLabel(chart: Pick<CommitteeDutiesChartRow, 'period_from' | 'period_to'>): string {
  const from = chart.period_from.slice(0, 10);
  const to = chart.period_to?.slice(0, 10);
  return to ? `${from} → ${to}` : `${from} → Ongoing`;
}

File: mobile/lib/models/committee_duty.dart
class CommitteeDutyRow {
  const CommitteeDutyRow({
    required this.id,
    required this.dutyLabel,
    required this.supervisorNames,
    this.sortOrder = 0,
  });

  final String id;
  final String dutyLabel;
  final List<String> supervisorNames;
  final int sortOrder;

  factory CommitteeDutyRow.fromRow(Map<String, dynamic> row) => CommitteeDutyRow(
        id: row['id'] as String,
        dutyLabel: row['duty_label'] as String,
        supervisorNames: (row['supervisor_names'] as List?)?.map((e) => e.toString()).toList() ?? const [],
        sortOrder: row['sort_order'] as int? ?? 0,
      );
}

class CommitteeDutiesChart {
  const CommitteeDutiesChart({
    required this.id,
    required this.periodFrom,
    this.periodTo,
    required this.rows,
  });

  final String id;
  final String periodFrom;
  final String? periodTo;
  final List<CommitteeDutyRow> rows;

  factory CommitteeDutiesChart.fromChartAndRows(
    Map<String, dynamic> chart,
    List<Map<String, dynamic>> rows,
  ) =>
      CommitteeDutiesChart(
        id: chart['id'] as String,
        periodFrom: chart['period_from'] as String,
        periodTo: chart['period_to'] as String?,
        rows: rows.map(CommitteeDutyRow.fromRow).toList(),
      );
}
